src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -74,16 +73,16 @@ public JsonWebToken(string jwtEncodedString)
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="JsonWebToken"/> class where the header contains the crypto algorithms applied to the encoded header and payload. The jwtEncodedString is the result of those operations.
+        /// Initializes a new instance of the <see cref="JsonWebToken"/> class where the header contains the crypto algorithms applied to the encoded header and payload.
         /// </summary>
-        /// <param name="header">Contains JSON objects representing the cryptographic operations applied to the JWT and optionally any additional properties of the JWT.</param>
-        /// <param name="payload">Contains JSON objects representing the claims contained in the JWT. Each claim is a JSON object of the form { Name, Value }.</param>
+        /// <param name="header">A string containing JSON which represents the cryptographic operations applied to the JWT and optionally any additional properties of the JWT.</param>
+        /// <param name="payload">A string containing JSON which represents the claims contained in the JWT. Each claim is a JSON object of the form { Name, Value }.</param>
         /// <exception cref="ArgumentNullException">'header' is null.</exception>
         /// <exception cref="ArgumentNullException">'payload' is null.</exception>
-        public JsonWebToken(JObject header, JObject payload)
+        public JsonWebToken(string header, string payload)
         {
-            Header = header ?? throw LogHelper.LogArgumentNullException(nameof(header));
-            Payload = payload ?? throw LogHelper.LogArgumentNullException(nameof(payload));
+            Header = JObject.Parse(header) ?? throw LogHelper.LogArgumentNullException(nameof(header));
+            Payload = JObject.Parse(payload) ?? throw LogHelper.LogArgumentNullException(nameof(payload));
```

Comment:
These null coalescing operators are likely dead code now. The JObject.Parse() method throws a JsonReaderException exception if the input contains invalid json. Otherwise, I would expect that it always returns a valid JObject reference.

Reply:
You're right. 👍 This isn't the right place for the null checks. I've changed this around so that the null checks are actually done on the header and payload parameters.